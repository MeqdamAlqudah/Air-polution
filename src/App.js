import { useSelector } from 'react-redux';
import {
  Routes,
  Route,
} from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import Continent from './components/Continent/Continent';
import CountriesList from './components/Countries/CountriesList';
import CountryAirData from './components/Countries/CountryAirData';

function App() {
  const countryData = useSelector((el) => el.countryName);
  return (
    <Routes>
      <Route
        path="/"
        element={(
          <Continent />

        )}
        exact
      />
      <Route
        path="/Europe"
        element={
          <CountriesList Continent="Europe" />
        }
      />
      <Route path="/SouthAmerica" element={<CountriesList Continent="South America" />} />
      <Route path="/NorthAmerica" element={<CountriesList Continent="North America" />} />
      <Route path="/Oceania" element={<CountriesList Continent="Oceania" />} />
      <Route path="/Asia" element={<CountriesList Continent="Asia" />} />
      <Route path="/Antarctica" element={<CountriesList Continent="Antarctica" />} />
      <Route path="/Africa" element={<CountriesList Continent="Africa" />} />
      <Route
        key={uuidv4()}
        path={`/${countryData.Continent}/Data`}
        element={(
          <CountryAirData
            name={countryData.name}
            continent={countryData.Continent}
            imgUrl={countryData.maps}
            lan={countryData.lan}
            long={countryData.long}
          />
        )}
      />

    </Routes>
  );
}

export default App;
