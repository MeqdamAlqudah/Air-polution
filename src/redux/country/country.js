const fetchData = 'countries/countries/fetchcountries';
const resetCountryData = 'countries/countries/reset';
const initialState = [];

export const getCountry = (payload) => ({
  type: fetchData,
  payload,

});

export const resetCountries = () => ({
  type: resetCountryData,
});
const reducer = (state = initialState, action) => {
  switch (action.type) {
    case fetchData:
      return [...Object.entries(action.payload).map((el) => el)];
    case resetCountryData:
      return [];
    default:
      return state;
  }
};

export default reducer;
