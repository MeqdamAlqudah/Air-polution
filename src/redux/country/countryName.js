const countryData = 'countries/countryName/changeTheCountry';
const resetCountryData = 'countries/countryName/reset';
const initialState = {};

export const getCountryData = (payload) => ({
  type: countryData,
  payload,

});

export const resetCountryName = () => ({
  type: resetCountryData,
});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case countryData:
      return action.payload;
    case resetCountryData:
      return {};
    default:
      return state;
  }
};

export default reducer;
