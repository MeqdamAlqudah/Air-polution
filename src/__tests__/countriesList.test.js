import React from 'react';
import renderer from 'react-test-renderer';
import {
  NavLink,
  BrowserRouter as Router,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import { getCountry } from '../redux/country/country';
import App from '../App';

describe('Test countriesList', () => {
  test('test NavLink path', async () => {
    const goBackHandler = jest.fn();
    const component = renderer.create(
      <Router>
        <NavLink onClick={goBackHandler} to="/" />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree.props.href).toEqual('/');
  });
  test('test country rendering ', () => {
    store.dispatch(getCountry({
      name: { common: 'Jordan' },
      population: 1000000,
      continent: ['Asia'],
    }));
    const component = renderer.create(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <App />

          </Router>
        </Provider>
      </React.StrictMode>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
