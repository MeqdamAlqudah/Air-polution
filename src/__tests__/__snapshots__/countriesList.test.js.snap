// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test countriesList test NavLink 1`] = `
<a
  aria-current="page"
  className="active"
  href="/"
  onClick={[Function]}
/>
`;

exports[`Test countriesList test NavLink 2`] = `
<a
  aria-current="page"
  className="active"
  href="/"
  onClick={[Function]}
/>
`;

exports[`Test countriesList test country rendering  1`] = `
<div
  className="container"
>
  <div
    className="worldMap"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      aria-controls="none"
      frameBorder="0"
      height="200"
      src="https://www.youtube.com/embed/e6rglsLy1Ys"
      title="YouTube video player"
      width="200"
    />
  </div>
  <h1
    className="h1"
  >
    Air Polution
  </h1>
  <ul
    className="ul"
  >
    <li
      className="firstContinent"
    >
      <a
        className=""
        href="/Europe"
        onClick={[Function]}
      >
        <img
          alt="europ"
          src="Europe.png"
        />
         
        <span>
          Europe
        </span>
        <p>
          44 coutry
        </p>
      </a>
    </li>
    <li
      className="secondContinent"
    >
      <a
        className=""
        href="/SouthAmerica"
        onClick={[Function]}
      >
        <img
          alt="South America"
          src="SouthAmerica.png"
        />
        <span>
          South America
        </span>
        <p>
          12 coutry
        </p>
      </a>
    </li>
    <li
      className="thirdContinent"
    >
      <a
        className=""
        href="/NorthAmerica"
        onClick={[Function]}
      >
        <img
          alt="North America"
          src="NorthAmerica.png"
        />
        <span>
          North America
        </span>
        <p>
          23 coutry
        </p>
      </a>
    </li>
    <li
      className="fourthContinent"
    >
      <a
        className=""
        href="/Oceania"
        onClick={[Function]}
      >
        <img
          alt="Oceania"
          src="Oceania.png"
        />
        <span>
          Oceania
        </span>
        <p>
          14 coutry
        </p>
      </a>
    </li>
    <li
      className="fifthContinent"
    >
      <a
        className=""
        href="/Asia"
        onClick={[Function]}
      >
        <img
          alt="Asia"
          src="Asia.png"
        />
        <span>
          Asia
        </span>
        <p>
          48 coutry
        </p>
      </a>
    </li>
    <li
      className="sixContinent"
    >
      <a
        className=""
        href="/Antarctica"
        onClick={[Function]}
      >
        <img
          alt="Antarctica"
          src="Antarctica.png"
        />
        <span>
          Antarctica
        </span>
        <p>
          5 coutry
        </p>
      </a>
    </li>
    <li
      className="sevenContinent"
    >
      <a
        className=""
        href="/Africa"
        onClick={[Function]}
      >
        <img
          alt="Africa"
          src="Africa.png"
        />
        <span>
          Africa
        </span>
        <p>
           54coutry
        </p>
      </a>
    </li>
  </ul>
</div>
`;
